athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaMetadataHandler.java

```diff
@@ -0,0 +1,386 @@
+            block.setValue("preparedStmt", rowNum, preparedSQLStmt) ? 1:0
+        );
+
+
+    }
+
+    /**
+     * Used to split-up the reads required to scan the requested batch of partition(s).
+     *
+     * Here we execute the SQL on Vertica
+     *
+     * @param allocator Tool for creating and managing Apache Arrow Blocks.
+     * @param request Provides details of the catalog, database, table, andpartition(s) being queried as well as
+     * any filter predicate.
+     * @return A GetSplitsResponse which primarily contains:
+     * 1. A Set<Split> which represent read operations Amazon Athena must perform by calling your read function.
+     * 2. (Optional) A continuation token which allows you to paginate the generation of splits for large queries.
+     */
+    @Override
+    public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest request)
+    {
+        //ToDo: implement use of a continuation token to use in case of larger queries
+
+        Connection connection = getConnection(request);
+        Set<Split> splits = new HashSet<>();
+        String exportBucket = System.getenv("export_bucket");
+        String queryId = request.getQueryId().replace("-","");
+
+        //get the SQL statement which was created in getPartitions
+        FieldReader fieldReader = request.getPartitions().getFieldReader("preparedStmt");
+        String sqlStatement  = fieldReader.readText().toString();
+        String catalogName = request.getCatalogName();
+
+        //execute the queries on Vertica
+        executeQueriesOnVertica(connection, sqlStatement);
+
+         /*
+          * For each generated S3 object, create a split and add data to the split.
+          */
+        Split split;
+        List<S3ObjectSummary> s3ObjectSummaries = getlistExportedObjects(exportBucket, queryId);
+        for (S3ObjectSummary objectSummary : s3ObjectSummaries)
+        {
+            split = Split.newBuilder(makeSpillLocation(request), makeEncryptionKey())
+                    .add("query_id", queryId)
+                    .add(VERTICA_CONN_STR, getConnStr(request))
+                    .add("exportBucket", exportBucket)
+                    .add("s3ObjectKey", objectSummary.getKey())
+                    .build();
+            splits.add(split);
+
+        }
+        logger.info("doGetSplits: exit - " + splits.size());
+        return new GetSplitsResponse(catalogName, splits);
+    }
+
+    /*
+     * Generates the necessary Prepared Statements to set the AWS Auth and export S3 bucket Region on Vertica
+     * and executes the queries
+     */
+    private void executeQueriesOnVertica(Connection connection, String sqlStatement)
+    {
+        try
+        {
+            //todo: find a way to work with STS credentials
+            //get the AWS Keys from Secrets Manager
+            String awsAccessId = resolveSecrets("${access_key}");
+            String awsSecretKey = resolveSecrets("${secret_key}");
```

Comment:
Does this mean the customer needs to have a Secret in Secretsmanager calls access_key? Perhaps this is just a placeholder for the STS idea you had?

Reply:
Removed this. The Instance Profiles of the EC2 instance hosting Vertica can be given permissions to the export S3 bucket